bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java

```diff
@@ -0,0 +1,198 @@
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.beam.sdk.extensions.gcp.util.GcsUtil;
+import org.apache.beam.sdk.extensions.gcp.util.gcsfs.GcsPath;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.snapshot.SnapshotTestingUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/*
+ * End to end integration test for pipeline that import HBase snapshot data into Cloud Bigtable.
+ * Prepare test data with gsutil(https://cloud.google.com/storage/docs/quickstart-gsutil):
+ * gsutil -m cp -r <PATH_TO_REPO>/bigtable-dataflow-parent/bigtable-beam-import/src/test/data/ \
+ *  gs://<test_bucket>/integration-test/
+ *
+ * Setup GCP credential: https://cloud.google.com/docs/authentication
+ *  Ensure your credential have access to Bigtable and Dataflow
+ *
+ * Run with:
+ * mvn integration-test -PhbasesnapshotsIntegrationTest \
+ * -Dgoogle.bigtable.project.id=<project_id> \
+ * -Dgoogle.bigtable.instance.id=<instance_id> \
+ * -Dgoogle.dataflow.stagingLocation=gs://<test_bucket>/staging \
+ * -Dcloud.test.data.folder=gs://<test_bucket>/integration-test/
+ */
+public class EndToEndIT {
+
+  private static final String TEST_SNAPSHOT_NAME = "test-snapshot";
+  // Location of test data hosted on Google Cloud Storage, for on-cloud dataflow tests.
+  private static final String CLOUD_TEST_DATA_FOLDER = "cloud.test.data.folder";
+  private static final String DATAFLOW_REGION = "region";
+
+  // Column family name used in all test bigtables.
+  private static final String CF = "cf";
+
+  // Full path of the Cloud Storage folder where dataflow jars are uploaded to.
+  private static final String GOOGLE_DATAFLOW_STAGING_LOCATION = "google.dataflow.stagingLocation";
+
+  private Connection connection;
+  private String projectId;
+  private String instanceId;
+  private String tableId;
+  private String region;
+
+  private GcsUtil gcsUtil;
+  private String dataflowStagingLocation;
+  private String workDir;
+  private byte[][] keySplits;
+
+  // Snapshot data setup
+  private String hbaseSnapshotDir;
+  private String restoreDir;
+
+  @Before
+  public void setup() throws Exception {
+    projectId = getTestProperty(BigtableOptionsFactory.PROJECT_ID_KEY);
+    instanceId = getTestProperty(BigtableOptionsFactory.INSTANCE_ID_KEY);
+    dataflowStagingLocation = getTestProperty(GOOGLE_DATAFLOW_STAGING_LOCATION);
+    region = getTestProperty(DATAFLOW_REGION);
+    String cloudTestDataFolder = getTestProperty(CLOUD_TEST_DATA_FOLDER);
+    if (!cloudTestDataFolder.endsWith(File.separator)) {
+      cloudTestDataFolder = cloudTestDataFolder + File.separator;
+    }
```

Comment:
For tests, I think it would be better to enforce a single format

Reply:
Noted, Will leave this one for another PR for re-structuring the test data and bucket lifecycle management.